nuitka/OptionParsing.py

```
@@ -1871,11 +2035,17 @@ def filterByName(key):
 
         option_name = "--%s" % key
 
-        if parser.isBooleanOption("--%s" % key):
+        if parser.isBooleanOption(option_name):
             if value == "false":
                 continue
 
             options_added.append(option_name)
+        elif parser.isListOption(option_name):
+            for value in value.split("\n"):
```

Ruff (0.8.2)

Loop control variable `value` overrides iterable it iterates

(B020)